Formula/caf.rb

```diff
@@ -27,11 +27,11 @@ class Caf < Formula
   fails_with gcc: "5"
 
   def install
-    mkdir "build" do
-      system "cmake", "..", *std_cmake_args, "-DCAF_ENABLE_TESTING=OFF"
-      system "make"
-      system "make", "install"
-    end
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args,
+                    "-DCMAKE_INSTALL_RPATH=#{lib}",
```

Comment:
Seen in #104139
```
==>FAILED
Full linkage --cached --test --strict caf output
  Files with missing rpath:
    /opt/homebrew/Cellar/caf/0.18.5/lib/libcaf_io.0.18.5.dylib
    /opt/homebrew/Cellar/caf/0.18.5/lib/libcaf_openssl.0.18.5.dylib
    /opt/homebrew/Cellar/caf/0.18.5/share/caf/tools/caf-run
    /opt/homebrew/Cellar/caf/0.18.5/share/caf/tools/caf-vec
```

Binaries in `pkgshare` need RPATH to `lib`, so can't use relative `#{rpath}`:
```console
❯ otool -L /usr/local/Cellar/caf/0.18.5/share/caf/tools/*
/usr/local/Cellar/caf/0.18.5/share/caf/tools/caf-run:
	@rpath/libcaf_io.0.18.5.dylib (compatibility version 0.18.5, current version 0.18.5)
	@rpath/libcaf_core.0.18.5.dylib (compatibility version 0.18.5, current version 0.18.5)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1200.3.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
/usr/local/Cellar/caf/0.18.5/share/caf/tools/caf-vec:
	@rpath/libcaf_core.0.18.5.dylib (compatibility version 0.18.5, current version 0.18.5)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1200.3.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)

❯ fd libcaf_core.0.18.5.dylib /usr/local/Cellar/caf/0.18.5/
/usr/local/Cellar/caf/0.18.5/lib/libcaf_core.0.18.5.dylib

❯ fd libcaf_io.0.18.5.dylib /usr/local/Cellar/caf/0.18.5/
/usr/local/Cellar/caf/0.18.5/lib/libcaf_io.0.18.5.dylib
```